dashboard/test/models/activity_test.rb

```diff
@@ -1,7 +1,79 @@
 require 'test_helper'
 
+require 'securerandom'
+require 'sqs/synchronous_queue'
+require 'timecop'
+
 class ActivityTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+
+  # A test handler which allows deferred handling of asynchronous messages.
+  # It records the messages it is alled with an then calls the underlying
+  # AsyncActivityHandler when `run_queued_messages` is called.
+  class DeferredActivityHandler
+    def initialize
+      @messages = []
+      @activity_handler = Activity::AsyncHandler.new
+    end
+
+    def handle(messages)
+      @messages += messages
```

Comment:
I think you can use <<  according to http://ruby-doc.org/core-2.2.0/Array.html#class-Array-label-Adding+Items+to+Arrays

On the same page is this comment:

Note that
x += y
is the same as
x = x + y
This means that it produces a new array. As a consequence, repeated use of += on arrays can be quite inefficient.